nf_core/pipeline-template/lib/NfcoreSchema.groovy

```diff
@@ -126,36 +132,36 @@ class NfcoreSchema {
 
         //=====================================================================//
         // Validate parameters against the schema
-        InputStream inputStream = new File(jsonSchema).newInputStream()
-        JSONObject rawSchema = new JSONObject(new JSONTokener(inputStream))
+        InputStream input_stream = new File(getSchemaPath(workflow, schema_filename=schema_filename)).newInputStream()
+        JSONObject raw_schema = new JSONObject(new JSONTokener(input_stream))
 
         // Remove anything that's in params.schema_ignore_params
-        rawSchema = removeIgnoredParams(rawSchema, params)
+        raw_schema = removeIgnoredParams(raw_schema, params)
 
-        Schema schema = SchemaLoader.load(rawSchema)
+        Schema schema = SchemaLoader.load(raw_schema)
 
         // Clean the parameters
         def cleanedParams = cleanParameters(params)
 
         // Convert to JSONObject
         def jsonParams = new JsonBuilder(cleanedParams)
-        JSONObject paramsJSON = new JSONObject(jsonParams.toString())
+        JSONObject params_json = new JSONObject(jsonParams.toString())
 
         // Validate
         try {
-            schema.validate(paramsJSON)
+            schema.validate(params_json)
         } catch (ValidationException e) {
             println ''
             log.error 'ERROR: Validation of pipeline parameters failed!'
             JSONObject exceptionJSON = e.toJSON()
-            printExceptions(exceptionJSON, paramsJSON, log)
+            printExceptions(exceptionJSON, params_json, log)
             println ''
             has_error = true
         }
 
         // Check for unexpected parameters
         if (unexpectedParams.size() > 0) {
-            Map colors = Utils.logColours(params.monochrome_logs)
+            Map colors = NfcoreTemplate.logColours(params.monochrome_logs)
```

Comment:
Did the filename for this change again? :facepalm:
Is it `NfcoreTemplate` instead of `Utils` now?

Reply:
We have both `Utils.groovy` and `NfcoreTemplate.groovy` now as we agreed on Slack 😅 The `logColours` stuff is only used in either `NfcoreSchema.groovy` or in `NfcoreTemplate.groovy` which is why I put it there. In theory, people that only use the nf-core pipeline  template to create their pipelines (and not to fork here) can then do whatever they wish in terms of logging etc. I have noticed this in quite a few aftermarket nf-core pipelines already.

Reply:
Haha yeah I think I missed that 😁 
Alright sounds good I don't have strong feelings about that, was just confused because I was following the naming convention there on a few pipelines and have changed names a few times now 😅  but soon we have a template 😁